tests/MongoDB.Driver.Examples/DocumentationExamples.cs

```diff
@@ -1568,6 +1569,65 @@ public void Index_Example_2()
             result.Should().Be("cuisine_1_name_1");
         }
 
+        [Fact]
+        public void Causal_Consistency_Example_1()
+        {
+            RequireServer.Check().SupportsCausalConsistency();
+
+            string itemsCollectionName = "items";
+            var testDatabase = client.GetDatabase("test");
+            testDatabase.DropCollection(itemsCollectionName);
+
+            // Start Causal Consistency Example
+            using (var session1 = client.StartSession(new ClientSessionOptions { CausalConsistency = true }))
+            {
+                var currenDate = DateTime.Now.Date;
+                var items = client.GetDatabase("test", new MongoDatabaseSettings
+                {
+                    ReadConcern = ReadConcern.Majority,
+                    WriteConcern = new WriteConcern(WriteConcern.WMode.Majority, TimeSpan.FromMilliseconds(1000))
+                })
+                .GetCollection<BsonDocument>("items");
+
+                items.UpdateOne(session1, document =>
+                    document["sku"] == "111" && document["end"] == BsonNull.Value,
+                    Builders<BsonDocument>.Update.Set("end", currenDate));
+
+                items.InsertOne(session1, new BsonDocument
+                {
+                    { "sku", "nuts-111" },
+                    { "name", "Pecans" },
+                    { "start", currenDate }
+                });
+
+                using (var session2 = client.StartSession(new ClientSessionOptions { CausalConsistency = true }))
+                {
+                    session2.AdvanceClusterTime(session1.ClusterTime);
+                    session2.AdvanceOperationTime(session1.OperationTime);
+
+                    var items2 = client.GetDatabase("test", new MongoDatabaseSettings
+                    {
+                        ReadPreference = ReadPreference.Secondary,
+                        ReadConcern = ReadConcern.Majority,
+                        WriteConcern = new WriteConcern(WriteConcern.WMode.Majority, TimeSpan.FromMilliseconds(1000))
+                    })
+                    .GetCollection<BsonDocument>("items");
+
+                    foreach (var item in items2.Find(session2, document => document["end"] == BsonNull.Value).ToEnumerable())
```

Comment:
Don't use lambdas (for filter) with BsonDocument.

Introduce filter variable to simplify foreach line?